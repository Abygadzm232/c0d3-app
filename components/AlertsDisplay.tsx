import React, { useState, useEffect } from 'react'
import Alert from './Alert'
import { AlertData, DismissedAlerts } from '../@types/alerts'

type Props = {
  alerts: AlertData[]
  page?: string
}

const AlertsDisplay: React.FC<Props> = ({ alerts, page }) => {
  const [dismissedAlerts, onDismiss] = useState<DismissedAlerts>({})
  useEffect(() => {
    const localDismissedAlerts = localStorage.getItem('dismissedAlerts')
    if (localDismissedAlerts) {
      onDismiss(JSON.parse(localDismissedAlerts))
    }
  }, [])

  const dismissAlert = (id: string) => {
    onDismiss(prevDismissedAlerts => {
      const newDismissedAlerts = { ...prevDismissedAlerts, [id]: true }
      localStorage.setItem(
        'dismissedAlerts',
        JSON.stringify(newDismissedAlerts)
      )
      return newDismissedAlerts
    })
  }

  const widthClass = page === 'curriculum' ? 'col-12' : ''
  return (
    <div className={`alerts-container ${widthClass}`}>
      {alerts
        .filter(alert => !dismissedAlerts[alert.id as string])
        .map(alert => (
          <Alert key={alert.id} alert={alert} onDismiss={dismissAlert} />
        ))}
    </div>
  )
}

export default AlertsDisplay
