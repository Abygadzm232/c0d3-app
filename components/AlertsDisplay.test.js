import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import AlertsDisplay from './AlertsDisplay'

describe('Alerts Display Component', () => {
  const alerts = [
    {
      id: '0',
      text: 'Please upgrade your CLI client by running npm update c0d3.',
      type: 'urgent'
    },
    {
      id: '1',
      text: 'Set up your computer to submit challenges.',
      type: 'info',
      url:
        'https://www.notion.so/JS-0-Foundations-a43ca620e54945b2b620bcda5f3cf672#b45ed85a95e24c9d9fb784afb7a46bcc',
      urlCaption: 'View Instructions'
    }
  ]
  test('Should dismiss alert for curriculum page', () => {
    const { getAllByRole } = render(
      <AlertsDisplay alerts={alerts} page="curriculum" />
    )
    let displayedAlerts = getAllByRole('alert')
    expect(displayedAlerts.length).toEqual(2)
    const firstDismiss = getAllByRole('dismiss')[0]
    fireEvent.click(firstDismiss)
  })
})
