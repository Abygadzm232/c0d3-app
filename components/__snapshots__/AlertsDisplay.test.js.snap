// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts Display Component Should render properly when no props are passed in 1`] = `
<div>
  <div
    class="alerts-container "
  />
</div>
`;
